### ISSUE TITLE
vaadin-treegrid shows "[object Object]" tooltips when using MPR

### ISSUE THREAD

**Issue by roelnoten** [vaadin/tree-grid#41]
![image](https://github.com/vaadin/multiplatform-runtime/assets/14275880/d5a070d6-68d9-4383-b09f-eb5a17330d42)

Hi. We have an app on Vaadin 7.7.32 and have a lot of instances of treegrids (vaadin-treegrid addon 0.7.8).
Since we added MPR (version through bom) and Vaadin 23.3.2 we have weird tooltips in the treegrids showing "[object Object]".
All legacy views are wrapped in flow routes.
 
It can even be reproduced with simple code like
```
HierarchicalContainer container = new HierarchicalContainer();
container.addContainerProperty("1", String.class, null);
container.addContainerProperty("2", String.class, null);
TreeGrid g = new TreeGrid();
g.setContainerDataSource(container);
g.setWidth("100%");
g.setHeight("300px");
g.addRow("aaa", "bbb");
```
 
or
 
```
HierarchicalContainer cont = new HierarchicalContainer();
cont.addContainerProperty("aa", String.class, null);
cont.addContainerProperty("bb", String.class, null);
Item item = cont.addItem("numbers");
item.getItemProperty("aa").setValue("first");
item.getItemProperty("bb").setValue("second");
TreeGrid grid = new TreeGrid();
grid.setContainerDataSource(cont);
grid.setWidth("100%");
grid.setHeight("300px");
```
When setting a cell tooltip ourselves, that one is taken, so that replaces the ugly "[object Object]": g.setCellDescriptionGenerator((Grid.CellDescriptionGenerator) cell -> "cell");
When our setCellDescriptionGenerator tooltip returns null or "", it displays [object Object] again.....
 
We did not have this behaviour before the MPR & Vaadin 23 addition. I looked through https://github.com/vaadin/tree-grid/ but could not find what could cause this. Nothing is logged in the browser.
It is really the v-tooltip that is displaying the text.

**Comment by mcollovati** [comment]
The issue was triaged and currently added to the backlog priority queue for further investigation

**Comment by TatuLund** [comment]
Did you see the same problem with regular Vaadin 7 Grid? TreeGrid extends Grid both server and widget side and the tooltip mechanism is fully inherited. Thus I would like to know.

**Comment by roelnoten** [comment]
Extra info: the issue does not exist when using just a Grid. So it is really some interaction of the TreeGrid with MPR....

**Comment by mcollovati** [comment]
I was able to reproduce the issue on the tree grid demo project, by using the latest Vaadin 7.7 snapshot, but not with the latest stable 7.7.40.

@TatuLund could it be related to the usage of GWT 2.8 instead of 2.7?

**Comment by mcollovati** [comment]
A wild guess.

`GridConnector.getTooltipInfo()` has the following code
```java
            } else if (row.hasKey(GridState.JSONKEY_ROWDESCRIPTION)) {
                return createCellTooltipInfo(
                        row.getString(GridState.JSONKEY_ROWDESCRIPTION),
                        getState().rowTooltipContentMode);
            } else {
```

but in `HierarchyDataGenerator.generateData()` I can see

```java
        rowData.put(GridState.JSONKEY_ROWDESCRIPTION,
                JsonCodec.encode(hierarchyData, null, HierarchyData.class, getUI().getConnectorTracker())
                        .getEncodedValue());
```

Could it be possible that with GWT 2.7 `row.getString(GridState.JSONKEY_ROWDESCRIPTION)` silently failed or returned null, whereas with 2.8 it is returning the "toString" of the `JsonObject` that represents the `HierarchyData` POJO instance?

**Comment by TatuLund** [comment]
It is plausible explanation.

**Comment by TatuLund** [comment]
This indeed was a latent bug in TreeGrid, that was hidden due bug in older version of GWT. The GridState.JSONKEY_ROWDESCRIPTION should not have been used as key for the internal meta info in the first place. I fixed this by adding dedicated key. The fix is in version 0.7.9, which I just released.

**Comment by roelnoten** [comment]
Thank you Marco and Tatu, 
0.7.9 indeed fixes the issue. Thanks for the very fast resolution!
